realm/realm-library/src/main/java/io/realm/RealmResults.java

```diff
@@ -61,7 +49,8 @@
  * @see RealmQuery#findAll()
  * @see Realm#executeTransaction(Realm.Transaction)
  */
-public class RealmResults<E extends RealmModel> extends OrderedRealmCollectionImpl<E> {
+public class RealmResults<E extends RealmModel> extends OrderedRealmCollectionImpl<E>
+        implements RealmCollectionObservable<RealmResults<E>, OrderedRealmCollectionChangeListener<RealmResults<E>>> {
```

Comment:
It implements `RealmCollectionObservable<T, S extends OrderedRealmCollectionChangeListener>`. Or I have some misunderstanding?